nxc/protocols/ssh.py

```diff
@@ -63,124 +77,219 @@ def create_conn_obj(self):
         # but that might be too much of an opsec concern - maybe add in a flag to do more checks?
-        stdin, stdout, stderr = self.conn.exec_command("id")
-        if stdout.read().decode("utf-8").find("uid=0(root)") != -1:
-            self.logger.info(f"Determined user is root via `id` command")
-            self.admin_privs = True
-            return True
-        stdin, stdout, stderr = self.conn.exec_command("sudo -ln | grep 'NOPASSWD: ALL'")
-        if stdout.read().decode("utf-8").find("NOPASSWD: ALL") != -1:
-            self.logger.info(f"Determined user is root via `sudo -ln` command")
-            self.admin_privs = True
-            return True
+        self.logger.info(f"Determined user is root via `id ; sudo -ln` command")
+        stdin, stdout, stderr = self.conn.exec_command("id ; sudo -ln 2>&1")
+        stdout = stdout.read().decode("utf-8", errors="ignore")
+        admin_flag = {
+            "(root)": [True, None], 
+            "NOPASSWD: ALL": [True, None],
+            "(ALL : ALL) ALL": [True, None],
+            "(sudo)": [False, f'Current user: "{self.username}" was in "sudo" group, please try "--sudo-check" to check if user can run sudo shell']
+        }
+        for keyword in admin_flag.keys():
+            match = re.findall(re.escape(keyword), stdout)
+            if match:
+                self.logger.info(f'User: "{self.username}" matched keyword: {match[0]}')
+                self.admin_privs = admin_flag[match[0]][0]
+                if not self.admin_privs:
+                    tips = admin_flag[match[0]][1]
+                    continue
+                else:
+                    break
+        if not self.admin_privs and "tips" in locals():
+            self.logger.display(tips)
+        return
+
+    def check_if_admin_sudo(self):
+        if not self.password:
+            self.logger.error("Check admin with sudo not support private key.")
+            return
+        
+        if self.args.sudo_check_method:
+            method = self.args.sudo_check_method
+            self.logger.info(f"Doing sudo check with method: {method}")
+           
+        if method == "sudo-stdin":
+            stdin, stdout, stderr = self.conn.exec_command("sudo --help")
+            stdout = stdout.read().decode("utf-8", errors="ignore")
+            if "stdin" in stdout:
+                shadow_Backup = f'/tmp/{uuid.uuid4()}'
+                # sudo support stdin password
+                stdin, stdout, stderr = self.conn.exec_command(f"echo {self.password} | sudo -S cp /etc/shadow {shadow_Backup} >/dev/null 2>&1 &")
+                stdin, stdout, stderr = self.conn.exec_command(f"echo {self.password} | sudo -S chmod 777 {shadow_Backup} >/dev/null 2>&1 &")
+                tries = 1
+                while True:
+                    self.logger.info(f"Checking {shadow_Backup} if it existed")
+                    stdin, stdout, stderr = self.conn.exec_command(f'ls {shadow_Backup}')
```

Comment:
If you only need `stderr` for example like in this case please "throw away" the other variables with `_`, this increases readability a lot because you know now what to focus on (and is a bit more efficient i think). Here i think it would be `_, _, stderr = self.con...`

Reply:
Yes yes, you're right, should paste for anonymous variable